packages/core-data/src/hooks/use-entity-records.ts

```diff
@@ -152,3 +153,61 @@ export function __experimentalUseEntityRecords(
 	} );
 	return useEntityRecords( kind, name, queryArgs, options );
 }
+
+export function useEntityRecordsWithPermissions< RecordType >(
+	kind: string,
+	name: string,
+	queryArgs: Record< string, unknown > = {},
+	options: Options = { enabled: true }
+): EntityRecordsResolution< RecordType > {
+	const entityConfig = useSelect(
+		( select ) => select( coreStore ).getEntityConfig( kind, name ),
+		[ kind, name ]
+	);
+	const { records: data, ...ret } = useEntityRecords(
+		kind,
+		name,
+		queryArgs,
+		options
+	);
+	const ids = useMemo(
+		() =>
+			data?.map(
+				// @ts-ignore
+				( record: RecordType ) => record[ entityConfig?.key ?? 'id' ]
+			) ?? [],
+		[ data, entityConfig?.key ]
+	);
+
+	// Todo: this selector is not memoized properly.
```

Comment:
Thinking of moving the logic here into a private selector to be able to memoize it properly.

Reply:
Yeah, it probably would be easier to memoize as a separate private selector - `getEntityRecordsWithPermissions`.

Reply:
Updated to use a private selector. The private selector is not perfect as it's recomputed when the whole permissions object changes but if we are to optimize this we need to rework the reducer shape to be per kind/name...